internal/agent/agent.go

```diff
@@ -19,6 +21,17 @@ import (
 
 // Run starts the agent provider emitting the bootstrap event.
 func Run(opts ...Option) error {
+	// capture ctrl+c and exit cleanly
+	channel := make(chan os.Signal, 1)
+	signal.Notify(channel, os.Interrupt)
+	go func() {
```

Comment:
this will spawn up a goroutine and grow indefinetly on each time `Run` is called again